### PR TITLE
Cosmos Spark: Add option to use raw json with patch

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description

This PR adds the ability to patch via Spark when the DF has a property with raw json (instead of a StructType or ArrayType for example)

Whether or not the to be patched column contians raw json or not can be determined based on whether the `spark.cosmos.write.patch.columnConfigs` column conifguration has a suffix of `.rawJson`.

The config belog for the "car" column would expect that the DF contains rawJson in the "car" column
`[col(car).op(replace).rawJson]`

The default is that a schematized value exists in the DF. So, below config would expect the "car" column to contain a StructType, ValueType like StringType or ArrayType.
`[col(car).op(replace)]`

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ### 4.16.0-beta.1 (Unreleased)
 
 #### Features Added
+* Added an option to use raw json when applying partial updates via WriteStrategy `ItemPatch`- See [PR 32610](https://github.com/Azure/azure-sdk-for-java/pull/32610)
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md:6]: Do we also need to add the documentation on how the rawJson will be identified (like what you mentioned in the PR description) ?
Or is it all implementation details that are not worth mentioning?

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala [hunk 2]
```diff
@@ -718,9 +719,9 @@ private object CosmosWriteConfig {
              // col[(](.*?)[)]: column name match
              // ([.]path[(](.*)[)])*: mapping path match, it is optional
              // [.]op[(](.*)[)]: patch operation mapping
-             val operationConfigaRegx = """(?i)col[(](.*?)[)]([.]path[(](.*)[)])*[.]op[(](.*)[)]$""".r
+             val operationConfigaRegx = """(?i)col[(](.*?)[)]([.]path[(](.*)[)])*[.]op[(](.*)[)](.rawJson$|$)""".r
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala:722]: With now .rawJson matching, we should update the above comments to reflect this change. Would help other developers to understand the rational behind this.

**Review by kushagraThapar — APPROVED** [review]
LGTM, thanks @FabianMeiswinkel

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md:6]: Right now we only have Annie's PDF document (which I can't really change) - but, yes - we also should add documentation like Theo added for throughput control. Will bring it up in next Java sync-up - at least the PR is something we can point folks to. And this change is to unblock a customer scenario

**Pull request merged**